executor/logger/worker.go

```diff
@@ -21,14 +22,31 @@ const (
 	InferenceServiceNameAttr = "inferenceservicename"
 	NamespaceAttr            = "namespace"
 	EndpointAttr             = "endpoint"
+	KafkaTypeHeader          = "type"
+	KafkaContentTypeHeader   = "content-type"
 )
 
 // NewWorker creates, and returns a new Worker object. Its only argument
 // is a channel that the worker can add itself to whenever it is done its
 // work.
-func NewWorker(id int, workerQueue chan chan LogRequest, log logr.Logger, sdepName string, namespace string, predictorName string) Worker {
+func NewWorker(id int, workerQueue chan chan LogRequest, log logr.Logger, sdepName string, namespace string, predictorName string, kafkaBroker string, kafkaTopic string) (*Worker, error) {
+
+	var producer *kafka.Producer
+	var err error
+	if kafkaBroker != "" {
+		log.Info("Creating producer", "broker", kafkaBroker, "topic", kafkaTopic)
+		producer, err = kafka.NewProducer(&kafka.ConfigMap{
+			"bootstrap.servers":   kafkaBroker,
+			"go.delivery.reports": false, // Need this othewise will get memory leak
```

Comment:
Ok interesting, why is this the case? Is it the issue that it had reaching the memory limits?

Reply:
Due to https://github.com/confluentinc/confluent-kafka-go/issues/341